dotnet/src/SemanticKernel.Abstractions/AI/ChatCompletion/IChatCompletion.cs

```diff
@@ -48,4 +48,21 @@ public interface IChatCompletion : IAIService
         string prompt,
         PromptExecutionSettings? executionSettings = null,
         CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Get streaming results for the prompt using the specified request settings.
+    /// Each modality may support for different types of streaming result.
+    /// </summary>
+    /// <remarks>
+    /// Usage of this method may be more efficient if the connector has a dedicated API to return this result without extra allocations for StreamingResultChunk abstraction.
+    /// </remarks>
+    /// <exception cref="NotSupportedException">Throws if the specified type is not the same or fail to cast</exception>
+    /// <param name="chatHistory">The chat history to complete.</param>
+    /// <param name="executionSettings">Request settings for the completion API</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>Streaming list of different completion streaming string updates generated by the remote model</returns>
+    IAsyncEnumerable<T> GetStreamingContentAsync<T>(
+        ChatHistory chatHistory,
```

Comment:
Are convenient when you don't have a chat history, but I could make it an extension instead of an interface one.

This was one excercise towards the Multi modality work.

Reply:
Looking in more detail, the ChatCompletion that also supports the ITextCompletion would have the string input as one of the API's.